integrations/signin_test.go

```diff
@@ -0,0 +1,53 @@
+// Copyright 2017 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package integrations
+
+import (
+	"net/http"
+	"testing"
+
+	"github.com/Unknwon/i18n"
+	"github.com/stretchr/testify/assert"
+)
+
+func testLoginFailed(t *testing.T, username, password, errorMessage string) *TestResponse {
+	session := emptyTestSession(t)
+
+	req := NewRequestWithValues(t, "POST", "/user/login", map[string]string{
+		"_csrf":     GetCSRF(t, session, "/user/login"),
+		"user_name": username,
+		"password":  password,
+	})
+	resp := session.MakeRequest(t, req, http.StatusUnauthorized)
+
+	htmlDoc := NewHTMLParser(t, resp.Body)
+	resultMsg := htmlDoc.doc.Find(".ui.message>p").Text()
+
+	assert.EqualValues(t, errorMessage, resultMsg)
+
+	return resp
+}
+
+func TestSignin(t *testing.T) {
+	samples := []struct {
+		username     string
+		password     string
+		errorMessage string
+	}{
+		{username: "wrongUsername", password: "password", errorMessage: i18n.Tr("en", "form.username_password_incorrect")},
+		{username: "user15", password: "wrongPassword", errorMessage: i18n.Tr("en", "form.username_password_incorrect")},
+		{username: "wrongUsername", password: "wrongPassword", errorMessage: i18n.Tr("en", "form.username_password_incorrect")},
+		{username: "user15@example.com", password: "password", errorMessage: i18n.Tr("en", "form.email_been_used")},
+		{username: "user1@example.com", password: "wrongPassword", errorMessage: i18n.Tr("en", "form.username_password_incorrect")},
+	}
+
+	prepareTestEnv(t)
```

Comment:
if test-case would change data that need to be reset than yes, otherwise it will only make tests run slower.